pypika/terms.py

```diff
@@ -288,57 +288,111 @@ def get_sql(self, **kwargs: Any) -> str:
         raise NotImplementedError()
 
 
+def idx_placeholder_gen(idx: int) -> str:
+    return str(idx + 1)
+
+
+def named_placeholder_gen(idx: int) -> str:
+    return f'param{idx + 1}'
+
+
 class Parameter(Term):
     is_aggregate = None
 
     def __init__(self, placeholder: Union[str, int]) -> None:
```

Comment:
Type hint here doesn't include callable? Is that correct still?

Reply:
Correct, only ListParameter and DictParameter (and classes that inherit from them) support callable. 
The base parameter does not.